### PR TITLE
feat(docs): Migrate typehead docs

### PR DESCRIPTION
**Pull request by bry4n**
JIRA: https://jira.rax.io/browse/FRMW-206

*  [x] Dev LGTM
*  [x] Dev LGTM

### CODE REVIEW

**Comment by CITguy** [comment]
@bry4n, please rebase to correct Travis error.

**Comment by bry4n** [comment]
@CITguy rebased.

File: src/typeahead/typeahead.js [hunk 1]
```diff
@@ -4,7 +4,22 @@
...
  *
- * [TBD]
+ * This component provides styles and a demo for the
+ * [the Angular-UI Bootstrap Typeahead plugin](https://goo.gl/EMGTTq),
+ * which is included as a dependency for EncoreUI.
+ *
+ * ## Usage
+ *
+ * Usage is the exact same as demoed on the Angular-UI Bootstrap site. See
+ * [the Angular-UI Bootstrap Docs](http://angular-ui.github.io/bootstrap/) for
```
Comment (100stacks) [inline src/typeahead/typeahead.js]: Recommend pointing to the _Angular Typeahead_ directive section:
`http://angular-ui.github.io/bootstrap/#/typeahead`

File: src/typeahead/typeahead.js [hunk 2]
```diff
@@ -4,7 +4,22 @@
...
+ * further guidance on usage and configuration of this component.
+ *
+ * A feature has been added that shows the list of options when the input
+ * receives focus.  This list is still filtered according to the input's value,
+ * except when the input is empty.  In that case, all the options are shown.
+ * To use this feature, add the `allowEmpty` parameter to the `filter` filter
+ * in the `typeahead` attribute.  See the demo for an example.
```
Comment (CITguy) [inline src/typeahead/typeahead.js]: Please link to the component demo page.

```
See the {@link /#/components/typeahead demo} for an example.
```

Comment (bry4n) [inline src/typeahead/typeahead.js]: updated

File: src/typeahead/typeahead.js (hunk 1 above)
Comment (bry4n) [inline src/typeahead/typeahead.js]: updated

**Comment by CITguy** [comment]
LGTM

**Comment by 100stacks** [comment]
LGTM

**Pull request merged**